sktime/transformations/series/compose.py

```diff
@@ -91,8 +124,239 @@ def transform(self, Z, X=None):
+
+        # transform to pd.Dataframe in univariate case
+        is_series = False
+        if isinstance(z, pd.Series):
+            is_series = True
+            name_of_z = z.name
+            z = z.to_frame()
+
+        # make copy of z
+        z = z.copy()
+
+        # make sure z contains all columns that the user wants to transform
+        self._check_columns(z)
+        for colname in self.columns_:
+            z[colname] = self.transformers_[colname].transform(z[colname], X)
+
+        # make z a series again in univariate case
+        if is_series:
+            z = self._revert_to_series(name_of_z, z)
+        return z
+
+    @if_delegate_has_method(delegate="transformer")
+    def inverse_transform(self, Z, X=None):
+        """
+        Inverse-transform data.
+
+        Returns an inverse-transformed version of Z by iterating over specified
+        columns and applying the univariate series transformer to them.
+        Only works if self.transformer has an inverse-transform method.
+
+        Parameters
+        ----------
+        Z : pd.Series, pd.DataFrame
+
+        Returns
+        -------
+        Z : pd.Series, pd.DataFrame
+            Inverse-transformed time series(es).
+        """
+        self.check_is_fitted()
+        z = check_series(Z)
+
+        # make z a pd.Dataframe in univariate case
+        is_series = False
+        if isinstance(z, pd.Series):
+            name_of_z = z.name
+            z = z.to_frame()
+            is_series = True
+
+        # make copy of z
+        z = z.copy()
+
+        # make sure z contains all columns that the user wants to transform
+        self._check_columns(z)
+
+        # iterate over columns that are supposed to be inverse_transformed
+        for colname in self.columns_:
+            z[colname] = self.transformers_[colname].inverse_transform(z[colname], X)
+
+        # make z a series again in univariate case
+        if is_series:
+            z = self._revert_to_series(name_of_z, z)
+        return z
+
+    @if_delegate_has_method(delegate="transformer")
+    def update(self, Z, X=None, update_params=True):
+        """
+        Update Parameters.
+
+        Update the parameters of the estimator with new data
+        by iterating over specified columns.
+        Only works if self.transformer has an update method.
+
+        Parameters
+        ----------
+        Z : pd.Series
+            New time series
+        update_params : bool, optional (default=True)
+
+        Returns
+        -------
+        self : an instance of self
+        """
+        z = check_series(Z)
+
+        # make z a pd.DataFrame in univariate case
+        if isinstance(z, pd.Series):
+            z = z.to_frame()
+
+        # make sure z contains all columns that the user wants to transform
+        self._check_columns(z)
+        for colname in self.columns_:
+            self.transformers_[colname].update(z[colname], X)
+        return self
+
+    def _check_columns(self, z):
```

Comment:
this could be a stand-alone function into which you pass the `self.columns_` - the idea is that stand-alone functions are usually easier to understand than methods